src/test/groovy/org/prebid/server/functional/AuctionSpec.groovy

```diff
@@ -23,4 +33,113 @@ class AuctionSpec extends BaseSpec {
         BidRequest.defaultBidRequest || "valid bid request"
         new BidRequest()             || "invalid bid request"
     }
+
+    def "PBS should apply timeout from stored request when it's not specified in the auction request"() {
+        given: "Default basic BidRequest with generic bidder"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            tmax = null
+            ext.prebid.storedRequest = new PrebidStoredRequest(id: PBSUtils.randomNumber.toString())
+        }
+
+        and: "Default stored request with timeout"
+        def timeout = PBSUtils.getRandomNumber(0, MAX_TIMEOUT)
```

Comment:
This will fail miserably and unpredictably because on some runs you will get very small timeout. Also, use method from base class, that I mentioned earlier.